packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -148,15 +147,11 @@ function useBlockEditorSettings( settings, postType, postId, renderingMode ) {
 
 			function getSectionRootBlock() {
 				if ( renderingMode === 'template-locked' ) {
-					return getBlocksByName( 'core/post-content' )?.[ 0 ] ?? '';
+					return getBlocksByName( 'core/post-content' )?.[ 0 ];
```

Comment:
```suggestion
					return getBlocksByName( 'core/post-content' )?.[ 0 ] ?? ''
```

If we restore this then when renderingMode is `template-locked` and there is no `post-content` block, the algorithm to search for the `main` tag will not be used and the top level block will be selected.

Reply:
I don't think we should be searching for `main` when editing pages (or entities where the template is locked). That is likely defined elsewhere.

So if there's no post-content block we should probably return the empty string like we used to do.